File: python/resistor-color-duo/resistor_color_duo.py
def value(colors) -> int:
    return int(''.join([ str(value_single(_)) for _ in colors[:2] ]))

def value_single(color:str) -> int:
    return [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    ].index(color)

File: javascript/resistor-color-duo/resistor-color-duo.js
export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

function colorValue(color) {
  // assuming all color inputs are valid in COLORS
  return COLORS.indexOf(color);
}

export const decodedValue = (colors) => {
  // assuming colors has always more than two elements
  const [color1, color2] = colors;
  return colorValue(color1) * 10 + colorValue(color2);
}